### PR TITLE
reintroduce multiple uplooking camera support

### PR DESCRIPTION
**Pull request by cri-s**
This reintroduce the option to support multiple uplooking parallel cameras without the need of recompile the sources.

# Read This First
To submit a Pull Request for OpenPnP you must use this template or it will not be accepted. 

Be sure to review the [Developers Guide](https://github.com/openpnp/openpnp/wiki/Developers-Guide)
and the [Contributing Guidelines](https://github.com/openpnp/openpnp/blob/develop/CONTRIBUTING.md).

Make your Pull Request as small as possible. Only include one bug or feature.
Large pull requests that change dozens of files or add multiple features are unlikely to be accepted.

Fill out all the details below. All sections below are required. If they are not included your Pull Request
will not be accepted.

-----------------------------------------------------------------------

# Description
This reintroduce the feature of having multiple parallel uplooking cameras without need
for customizing the source.

# Justification
Remove the need to have custom version just for the parallel uplooking camera use.

# Instructions for Use
No functionaly change to existing users, 

This is documentation for a user, not for a developer.
users with parallel cameras need just exchange the camera inside vision pipeline using
scripting stage based on nozzle Id.

# Implementation Details
1. How did you test the change? Be descriptive. Untested code will not be accepted.
tested with 2-8 cameras
2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
yes
3. If you made changes in the `org.openpnp.spi` or `org.openpnp.model` packages you will need to add additional justification for these changes. Changes to these packages require extensive review and testing.
4. Be sure to run `mvn test` before submitting the Pull Request. If the tests do not pass the Pull Request will not be accepted.

### CODE REVIEW

**Comment by vonnieda** [comment]
Currently on vacation, will review in 1 week.

**Comment by vonnieda** [comment]
This change seems to be specific to your script. As this code is written, it does nothing without whatever script you are using. This is not a good general solution. 

If you want this merged, please post the scripts that it requires, along with instructions on how to use them.

**Comment by cri-s** [comment]
Here the script, it is a vision pipeline stage script intended to be run with ScriptRun Stage.
It have no stage arguments.
[cam.zip](https://github.com/openpnp/openpnp/files/2102476/cam.zip)

Here the small header inside the file reported here for convenience.
///////////////////////////////////////
// first machine cam need to have location set X and Y to NaN (need machine.xml edit)
// second machine cam need to have location set for default Nozzle on first Camera !!
// if going manually over cameras, use the second camera with default Nozzle !!
// small cams need to be named like nozzle, as example if  nozzle id is N2
// small camera for Nozzle N1 need to have the ID camN2 .
// last cam is generic bigger cam
///////////////////////////////////////
double max=12; // mm max component diameter
///////////////////////////////////////

This does sequencially vision. There exists script that instead launch 2/4/8 threads in parallel.
It depend on machine and gpu usage what is the fastest solution.
Using the china machines with multiple cameras, the sequencial solution must be used because it
is required to switch the crossswitch multiplexer inside the analog video grabber.

Basically the code change ensure that the correct pixel/mm is used for the calcs.
Even using the same cameras, but with camera calibration and optics/assembly variations,
it happens that it haves different  pixel/mm .

**Comment by vonnieda** [comment]
Thanks @cri-s, this looks good, so I'll go ahead and merge it. 

For future readers, the line of interest in the script is:

```
pipeline.setProperty("camera",cam);
```

This sets the pipeline's camera property at the end of the scrip so that the caller can get access to it.
